neevaco/neeva-ios
# Bug fixes to Wallet

- Avoid crashes when there is no network connection, and make sure to recreate web3 when we have network
- Align the url being seen through the subscription calling updateSession and what we use to compute trustSignal
- The above fixes an issue with navigation to a verified site from malicious site showing a malicious site view again
- Avoid calling updateSession if we dont have a wallet, this avoid setting the session to nil continously for non-wallet users
- Fix remove wallet not showing up confirmation

### How was this tested?

### Manual test cases
Go to clayfriends.app
Observe the malicious site view. Click "Navigate to clayfriends.io"
Make sure no extra malicious site views pop up

Disconnect network and restart app (This should create a nil web3 object)
Reconnect network and go to a dApp and make sure it connects correctly

### Screenshots and screen recordings
https://user-images.githubusercontent.com/83613310/156872482-2c3fb107-ffb4-4c94-94f4-3173c6c7be31.mp4

## Issues addressed
Fixes #2953
